src/solvers/flattening/bv_utils.cpp

```diff
@@ -488,20 +488,28 @@ bvt bv_utilst::shift(const bvt &src, const shiftt s, std::size_t dist)
 
     switch(s)
     {
-    case shiftt::LEFT:
+    case shiftt::SHIFT_LEFT:
       l=(dist<=i?src[i-dist]:const_literal(false));
       break;
 
-    case shiftt::ARIGHT:
+    case shiftt::SHIFT_ARIGHT:
       l=(i+dist<src.size()?src[i+dist]:src[src.size()-1]); // sign bit
       break;
 
-    case shiftt::LRIGHT:
+    case shiftt::SHIFT_LRIGHT:
       l=(i+dist<src.size()?src[i+dist]:const_literal(false));
       break;
 
+    case shiftt::ROTATE_LEFT:
+      l=(dist<=i?src[i-dist]:src[src.size()-dist+i]);
```

Comment:
`src.size()-(dist+i)%src.size()` to avoid out-of-bounds access?

Reply:
no need!

Reply:
I don't see why this would be unnecessary. `boolbvt::convert_shift` will pass in an arbitrarily large value of `dist`. Straight from the user input. Which guard am I missing?

Reply:
dist = 16, i = 1, src.size() = 8 seems possible to me.

Reply:
ah, ok, I had assumed that dist<size. But indeed, that isn't necessarily true. Dist may even be negative.

Reply:
It isn't.  The floating-point code (specifically conversion of float to int) has cases where the size of the two operands to shift are not equal in size (in both directions).